### PR TITLE
Add support for Savage & Ravage's growing and youth potions

### PR DESCRIPTION
**Pull request by Voliant**
The booflo can grow up, but does not extend `AgeableEntity` so it doesn't support S&R's potions by default. I've already added checks for the different booflo entities in particle creation methods on my end, but I'm pull requesting the actual growing into endergetic because it's much easier if i can create a `growDown` method and use `growUp` without indirection.

### CODE REVIEW

**Comment by Voliant** [comment]
If i was right in what I thought you meant, I've fixed it.

File: src/main/java/endergeticexpansion/core/events/SRCompatEvents.java [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+@Mod.EventBusSubscriber(modid = EndergeticExpansion.MOD_ID)
+public class SRCompatEvents {
+    @SubscribeEvent
+    public static void onPotionExpire(PotionEvent.PotionExpiryEvent event) {
+        LivingEntity affected = event.getEntityLiving();
+        boolean isBabyEffect = event.getPotionEffect().getPotion() == ForgeRegistries.POTIONS.getValue(new ResourceLocation("savageandravage:shrinking"));
+        if(isBabyEffect || event.getPotionEffect().getPotion() == ForgeRegistries.POTIONS.getValue(new ResourceLocation("savageandravage:growth"))) {
+            if(!isBabyEffect && affected instanceof EntityBoofloBaby) ((EntityBoofloBaby)affected).growUp();
+            if(affected instanceof EntityBoofloAdolescent) {
+                if (isBabyEffect) {
```
Comment (SmellyModder) [inline src/main/java/endergeticexpansion/core/events/SRCompatEvents.java:27]: This line here is the last nitpick 😛

Comment (Voliant) [reply src/main/java/endergeticexpansion/core/events/SRCompatEvents.java:27]: Are you suggesting to add spacing between the if statements? Sorry if I’m missing something, I’m in bed on mobile

Comment (Voliant) [reply src/main/java/endergeticexpansion/core/events/SRCompatEvents.java:27]: Ok i just went on the pc and i actually don't know what you're referring to

Comment (SmellyModder) [reply src/main/java/endergeticexpansion/core/events/SRCompatEvents.java:27]: Most of your if statements have no spacing between them, so all of them should do that.

Comment (Voliant) [reply src/main/java/endergeticexpansion/core/events/SRCompatEvents.java:27]: I see. Changing now.

Comment (SmellyModder) [reply src/main/java/endergeticexpansion/core/events/SRCompatEvents.java:31]: Not what I was referring to lol...

Comment (Voliant) [reply src/main/java/endergeticexpansion/core/events/SRCompatEvents.java:31]: Well, what were you referring to? There's spaces around the && and ||, and that's the only other thing I can think of. Can you please just write out what changes you want me to make?

**Comment by Voliant** [comment]
Ok the change as requested on Discord is now done.

**Comment by SmellyModder** [comment]
Looks good.

**Pull request merged**